posthog/api/test/test_feature_flag.py

```diff
@@ -352,6 +375,127 @@ def test_deleting_feature_flag(self):
             },
         )
 
+        historical_flags = HistoricalVersion.objects.filter(team_id=self.team.id, name="FeatureFlag")
+        self.assertEqual(len(historical_flags), 1)
+        self.maxDiff = None
+        self.assertEqual(
+            historical_flags[0].state,
+            {
+                "id": None,
+                "key": "potato",
+                "name": "",
+                "_state": None,
+                "active": True,
+                # this property is for soft delete, and false here because this test is for hard delete
+                "deleted": False,
+                "filters": {},
+                "team_id": self.team.id,
+                "created_at": "2021-08-25T22:09:14.252000+00:00",
+                "created_by_id": self.user.id,
+                "rollout_percentage": None,
+            },
+        )
+        self.assertEqual(historical_flags[0].action, "delete")
+
+    @freeze_time("2021-08-25T22:09:14.252Z")
+    def test_get_feature_flag_that_needs_importing(self):
+        """
+        The first time we load history for a feature flag that existed before starting history logging
+        Import its current state
+        """
+        new_user = User.objects.create_and_join(self.organization, "new_annotations@posthog.com", None)
+
+        instance = FeatureFlag.objects.create(team=self.team, created_by=self.user)
+        self.client.force_login(new_user)
+
+        response = self.client.get(f"/api/projects/{self.team.id}/feature_flags/{instance.pk}/history")
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        history = response.json()["results"]
+
+        expected = [
+            asdict(
+                HistoryListItem(
+                    email="history.hog@posthog.com",
+                    name="history hog",
+                    action="FeatureFlag_imported",
+                    detail={"id": str(instance.pk), "key": ""},
+                    created_at="2021-08-25T22:09:14.252000+00:00",
+                )
+            )
+        ]
+
+        self.assertEqual(
+            history, expected,
+        )
+
+    def test_get_feature_flag_history(self):
```

Comment:
Can we consider that history would look like in cases of schema evolution?

e.g. 1. add a new field for feature flags "multivariate_type" with a default 2. feature flag updated 3. view history. Should / would multivariate_type be included in our history even if unchanged?

Reply:
You can't (I don't think) automagically associate a migration and a model in code, but you can query migrations by date. So you could write code to show history differently if showing a particular model after the date of a particular migration

I _think_ in your example above we would show the new default field as being changed in the first change after its introduction because it would be in the after state and not the before state

I think the simplest (but most human error-able) solution is that a migration that introduces new state to a model which logs history needs to be followed with a migration that adds a historical version identifying that we have added data to the model